File: Easy/registry/meta_registryaziendeview/meta_registryaziendeview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_registryaziendeview
{
    public class Meta_registryaziendeview : Meta_easydata 
	{
        public Meta_registryaziendeview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "registryaziendeview") {
				Name = "Enti e Aziende";
			EditTypes.Add("aziende");
			ListingTypes.Add("aziende");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idreg"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "aziende": {
						DescribeAColumn(T, "registryclass_description", "Tipologia", nPos++);
						DescribeAColumn(T, "title", "Denominazione", nPos++);
						DescribeAColumn(T, "registry_cf", "Codice fiscale", nPos++);
						DescribeAColumn(T, "registry_p_iva", "Partita iva", nPos++);
						DescribeAColumn(T, "registry_active", "attivo", nPos++);
						DescribeAColumn(T, "ateco_codice", "Codice Classificazione Ateco", nPos++);
						DescribeAColumn(T, "ateco_title", "Titolo Classificazione Ateco", nPos++);
						DescribeAColumn(T, "naturagiur_title", "Natura Giuridica", nPos++);
						DescribeAColumn(T, "numerodip_title", "Numero di dipendenti", nPos++);
						DescribeAColumn(T, "nace_idnace", "Codice Codice NACE", nPos++);
						DescribeAColumn(T, "nace_activity", "Activity Codice NACE", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "aziende": {
						return "title asc ";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/registry/meta_registryaziendeview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_registryaziendeview() {
        MetaData.apply(this, ["registryaziendeview"]);
        this.name = 'meta_registryaziendeview';
    }

    meta_registryaziendeview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_registryaziendeview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'aziende':
						this.describeAColumn(table, 'title', 'Denominazione', null, 1000, 101);
						this.describeAColumn(table, 'registryclass_description', 'Tipologia', null, 2200, 150);
						this.describeAColumn(table, 'registry_cf', 'Codice fiscale', null, 4000, 16);
						this.describeAColumn(table, 'registry_p_iva', 'Partita iva', null, 5000, 15);
						this.describeAColumn(table, 'registry_active', 'attivo', null, 6000, null);
						this.describeAColumn(table, 'ateco_codice', 'Codice Classificazione Ateco', null, 7200, 50);
						this.describeAColumn(table, 'ateco_title', 'Titolo Classificazione Ateco', null, 7300, 255);
						this.describeAColumn(table, 'naturagiur_title', 'Natura Giuridica', null, 8200, 200);
						this.describeAColumn(table, 'numerodip_title', 'Numero di dipendenti', null, 9200, 50);
						this.describeAColumn(table, 'nace_idnace', 'Identificativo NACE', null, 10100, 50);
						this.describeAColumn(table, 'nace_activity', 'Activity NACE', null, 10200, -1);
						this.describeAColumn(table, 'geo_nation_title', 'Nazionalità', null, 11300, 65);
						this.describeAColumn(table, 'registry_pic', 'Participant Identification Code (PIC)', null, 12000, 10);
						this.describeAColumn(table, 'registry_flag_pa', 'Ente pubblico', null, 100000, null);
//$objCalcFieldConfig_aziende$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idreg"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "aziende": {
						return "title asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('registryaziendeview', new meta_registryaziendeview('registryaziendeview'));

	}());
